src/Fable.Transforms/Replacements.fs

```diff
@@ -2707,17 +2707,30 @@ let asyncs com (ctx: Context) r t (i: CallInfo) (_: Expr option) (args: Expr lis
     // Fable.Core extensions
     | meth -> Helper.CoreCall("Async", Naming.lowerFirst meth, t, args, i.SignatureArgTypes, ?loc=r) |> Some
 
-let guids (_: ICompiler) (ctx: Context) (_: SourceLocation option) t (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
+let guids (com: ICompiler) (ctx: Context) (r: SourceLocation option) t (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
     match i.CompiledName with
-    | "NewGuid"     -> Helper.CoreCall("String", "newGuid", t, []) |> Some
-    | "Parse"       -> Helper.CoreCall("String", "validateGuid", t, args, i.SignatureArgTypes) |> Some
-    | "TryParse"    -> Helper.CoreCall("String", "validateGuid", t, [args.Head; makeBoolConst true], [args.Head.Type; Boolean]) |> Some
-    | "ToByteArray" -> Helper.CoreCall("String", "guidToArray", t, [thisArg.Value], [thisArg.Value.Type]) |> Some
+    | "NewGuid"     -> Helper.CoreCall("Guid", "newGuid", t, []) |> Some
+    | "Parse"       -> Helper.CoreCall("Guid", "validateGuid", t, args, i.SignatureArgTypes) |> Some
+    | "TryParse"    -> Helper.CoreCall("Guid", "validateGuid", t, [args.Head; makeBoolConst true], [args.Head.Type; Boolean]) |> Some
```

Comment:
About your question whether we can make the validation at compile time for `Parse` and `Guid(...)`. Yes, you could check if args is `[Value (StringConstant aspiringGuid, _)]` and then do the validation at compile time on the `aspiringGuid`. Same for the `.ctor` below.

If a literal string is not passed as argument you can default to the runtime function as it's now.

Reply:
Thanks for the advice, it worked!